cmd/ipfs/main.go

```diff
@@ -181,15 +180,14 @@ func makeExecutor(req *cmds.Request, env interface{}) (cmds.Executor, error) {
 		exctr = client.(cmds.Executor)
 	} else {
 		cctx := env.(*oldcmds.Context)
-		pluginpath := filepath.Join(cctx.ConfigRoot, "plugins")
 
 		// check if repo is accessible before loading plugins
 		ok, err := checkPermissions(cctx.ConfigRoot)
 		if err != nil {
 			return nil, err
 		}
 		if ok {
-			if _, err := loader.LoadPlugins(pluginpath); err != nil {
+			if _, err := loader.LoadPlugins(cctx.ConfigRoot); err != nil {
```

Comment:
Hrm... making that code open up and parse the config on its own breaks some abstractions. The fact that the config is a file on disk is an implementation detail, We might need to rethink some things here.

cc @Stebalien @magik6k We need to get some configuration, and the peer ID to the tracing plugin. I'm not entirely sure yet how that should get fed through, ideas?

Reply:
For some context, here is how it may be used with the tracing plugin
https://github.com/ipfs/go-jaeger-plugin/pull/1/files#diff-e92ca16875e3c00b8a36338613029bf9R35

Reply:
This will also break changes in https://github.com/ipfs/go-ipfs/pull/4244/files#diff-153cb71153ee178fe2869148600ce8d5R197.

The problem is that at this stage we may not even have a peerID/config (i.e. user calling `ipfs init`), we may add some sort of late init for plugins that want it and call that above in `buildEnv`, either before or after calling `core.NewNode`

Reply:
For now setting the tracer name could be a manual process - maybe we don't want it to be the nodeID, I was doing this for convenience. 

This would still be done via a configuration file (where the tracer name is a field), but that file would reside in the plugin path - possibly under a directory based on the plugin name (e.g. `pluginPath/pluginName/config.cfg`).

I imagine we will want the ability to pass plugins config files in the future?